### DISCUSSION TITLE
How to resolve recursive ffmpeg <-> chromaprint dependency

### DISCUSSION THREAD

**Discussion by slhck** [Homebrew/discussions#5264] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.16-95-g9bd0367
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9bd03675f6cd0140e4dfad568efe58dc2224070e
Last commit: 10 hours ago
Core tap HEAD: 2e94172b28fbe3f5fc373cd5d3de759a2ab6ed76
Core tap last commit: 2 days ago
Core tap JSON: 05 Apr 05:52 UTC
Core cask tap HEAD: 6cd6dafa1aee1aae0f66f876f7ff123e412f1c22
Core cask tap last commit: 2 days ago
Core cask tap JSON: 05 Apr 05:52 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BOOTSNAP: set
HOMEBREW_CACHE: /Users/runner/Library/Caches/Homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_COLOR: set
HOMEBREW_CURL_PATH: /usr/bin/curl
HOMEBREW_CURLRC: 1
HOMEBREW_DEVELOPER: set
HOMEBREW_FAIL_LOG_LINES: 150
HOMEBREW_GIT_EMAIL: 1589480+BrewTestBot@users.noreply.github.com
HOMEBREW_GIT_NAME: BrewTestBot
HOMEBREW_GIT_PATH: /usr/bin/git
HOMEBREW_LOGS: /Users/runner/work/homebrew-ffmpeg/homebrew-ffmpeg/logs
HOMEBREW_MAKE_JOBS: 3
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_EMOJI: set
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSTALL_CLEANUP: set
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 3-core 64-bit dunno
Clang: 15.0.0 build 1500
Git: 2.39.3 => /usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.0.1 => /Applications/Xcode_15.0.1.app/Contents/Developer
Rosetta 2: false
```

### Output of `brew doctor`

```shell
(empty)
```

### Description of issue

We are maintaining a tap for ffmpeg, and for some reason we cannot run CI tests anymore, in a PR bumping ffmpeg to v7.0:

https://github.com/homebrew-ffmpeg/homebrew-ffmpeg/pull/160

The issue is this line: https://github.com/homebrew-ffmpeg/homebrew-ffmpeg/actions/runs/8565637192/job/23473994376?pr=160#step:3:103

    homebrew-ffmpeg/ffmpeg/ffmpeg depends on itself via chromaprint

Now, indeed the chromaprint formula [has an ffmpeg dependency](https://github.com/Homebrew/homebrew-core/blob/master/Formula/c/chromaprint.rb#L21). This has caused build failures for a while now but we don't know how to resolve it. Any suggestions would be appreciated!

**Comment by SMillerDev** [comment]
> This has caused build failures for a while now but we don't know how to resolve it. Any suggestions would be appreciated!

The only way is to break the recursive dependency. And since you can't control the chromaprint one, you'll have to remove the dependency on chromaprint from ffmpeg

**Reply by slhck** [reply]
But ffmpeg _can_ be built with chromaprint support and should support it.

From what I can tell chromaprint does not strictly require an ffmpeg library, and, in fact, has apparently removed some ffmpeg dependencies in v1.4.2: https://github.com/acoustid/chromaprint/releases

Per [its README](https://github.com/acoustid/chromaprint?tab=readme-ov-file#fft-library), you can swap out the ffmpeg library dependency for something else. Would this be a viable option? FFTW3 seems to be GPL-only; is that a problem for Homebrew?

**Reply by cho-m** [reply]
Based on https://github.com/acoustid/chromaprint/issues/129, there is no alternative for all features of `chromaprint` (i.e. `fpcalc`) and upstream recommendation is:
> If the circular dependency is a problem, I'd suggest disabling the chromaprint filter in FFmpeg. It's not a commonly used feature and people who really need it can build FFmpeg themselves. It's far more common for applications to use the fpcalc binary to generate audio fingerprints.
